metricbeat/helper/http_test.go

```diff
@@ -77,10 +77,11 @@ func TestGetAuthHeaderFromTokenNoFile(t *testing.T) {
 }
 
 func TestTimeout(t *testing.T) {
-	c := make(chan struct{})
-	defer close(c)
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		<-c
+		select {
+		case <-time.After(5 * time.Nanosecond):
```

Comment:
5 nanoseconds?

Reply:
Timeout is set to 1 nanosecond, this is just a higher time.